File: BankingSystemCSharp/Models/TransactionCommand.cs
public class CommandData
{
    public string AccountId { get; set; } = string.Empty;
    public decimal Amount { get; set; }
    public string? AccountName { get; set; }
    public string TargetAccountId { get; set; } = string.Empty;
}

public class TransactionCommand
{
    public string ActionType { get; set; }
    public CommandData commandData { get; set; }
}

File: BankingSystemRust/src/transaction_command.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize)]
pub struct TransactionCommand {
    #[serde(rename = "ActionType")]
    pub action_type: String,
    #[serde(rename = "commandData")]
    pub command_data: CommandData,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct CommandData {
    #[serde(rename = "AccountId")]
    pub account_id: String,
    #[serde(rename = "AccountName")]
    pub account_name: Option<String>,
    #[serde(rename = "Amount")]
    pub amount: f64,
    #[serde(rename = "TargetAccountId")]
    pub target_account_id: Option<String>,
}
